### PR TITLE
Problem 4 (WIP)

### LINKED ISSUE #10: Solve others/problem-4 in golang
Write test cases too

### PR DESCRIPTION
**Pull request by jmhobbs**
Closes #10 

Solver is working, but parser is failing on the very large inputs.  Working on sorting that out now.

### CODE REVIEW

File: others/problem-4/main.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func main() {
```
Comment (princerachit) [inline others/problem-4/main.go]: please remove main method. All the testing should be done via test files.

**Review by princerachit — CHANGES_REQUESTED** [review]

File: others/problem-4/main.go (hunk 1 above)
Comment (jmhobbs) [reply others/problem-4/main.go]: I added that to run it on the [hackerearth system](https://www.hackerearth.com/practice/algorithms/dynamic-programming/introduction-to-dynamic-programming-1/practice-problems/algorithm/xsquare-and-two-arrays/).  If you're not concerned with having it work on there than I'll drop `main` and this should be good to go.

Comment (princerachit) [reply others/problem-4/main.go]: Yeah please drop this.

**Comment by jmhobbs** [comment]
@princerachit removed the main.  Tests pass, should be good to go 👍

**Comment by princerachit** [comment]
Thanks for the pr mate. Feel free to create issues and work on them on your own.

**Pull request merged**